### ISSUE TITLE
Customize middle click action

### ISSUE THREAD

**Issue by t00** [nsz32/docklike-plugin#43]
I feel this might be a lightweight feature in line with docklite goals - it is awedomely simple btw.

Old habits die hard and I very much liked Windows taskbar feature to launch another application with a middle click on it's icon. docklite does what most managers do - closes the window - which although useful, I usually handle by Ctrl+W or Alt+F4 to save on the mouse mileage.

Please consider a simple dropdown in options, which will default to "Close All" as it is now, with additional options "Close", "Launch another", "Toggle Minimise", maybe "Preferences" although available only for supporting apps. All the features are already in code, hopefully this will be simple to do.

The workaround of holding Shift + Click on the icon will suffice for the time being :)

**Comment by nsz32** [comment]
Your feature request is next :)
note : If "Launch another" is selected, SHIFT+Click should close all windows

**Comment by t00** [comment]
That is absolutely brilliant, thank you. Btw since discovering docklite I never looked back, it should become a default xfce launcher.

**Comment by nsz32** [comment]
Docklite ? With the comment you made on it, I hope it was a typo, not a concurrent project :p
I will probably implement your feature next sunday.
Ty,

**Comment by gsantner** [comment]
I'm used to "middle click = open (new) instance of application".  
I just killed a application with important stuff - in the though it runs a second instance 😞

**Comment by nsz32** [comment]
Windows is a bad bad thing ...
I will not have time as expected to implement this request tomorrow.
You will have to wait some more times.

**Comment by t00** [comment]
That is all right, patience is a virtue.
Windows has its problems, lots of bloat, legacy UI and libs but also great ideas - as expected from 40 years old OS on which 1000s of devs have worked.

**Comment by nsz32** [comment]
Seven was quite cool.
But I didn't think I could work on 10 : it's becoming an android-with-windows-and-apparently-fancy-but-bloated-devil-stuff-cortana-powershell-click-click-updates. Won't pay (money and time) for that.

**Comment by t00** [comment]
Win10 is the reason I went full Linux 2 years ago

**Comment by nsz32** [comment]
That's a nice feature from Win10 :)
Note that a middle click closes tabs in many softwares, that wasn't a self-centric choice from me.

**Comment by t00** [comment]
Fair enough, this forced me to learn/set up shortcuts i.e. ctrl shift t for a new terminal window but from UX point of view middle click on app icon opening a new instance is more optimal as keyboard shortcut closing any app is always the same while there is no need to set up new instance shortcuts for all apps launched frequently, considering user is fine with using mouse.

**Comment by gsantner** [comment]
FYI, these are the options at the DockbarX dock, allowing to control close/maximize/newinstance etc individually per mouse button and modifier.

I do not request that, but just to let you know whats possible :D.

![Screenshot_20201228-212820](https://user-images.githubusercontent.com/6735650/103241511-b99b2d80-4953-11eb-9213-3e51246996f3.png)

By the way, everything with Windows and what people prefere etc is kinda offtopic? After all it's about being able to control the middle click action, not who likes which OS and who not 😄 .

**Comment by t00** [comment]
@gsantner indeed, in the end ability to work the way user prefers matters the most, although taking into consideration most widely adopted implementations and muscle memory is the key for wider adoption of Linux which is (almost) universally understood as a good cause. Complaining is off-topic...

Personally I am using XFCE with docklike instead of the. industry. standard. Gnome 3 (now 4). It is because of lack of ability to configure simple things without installing obscure plugins. It's tablet-like application launcher is conflicting with the idea of a desktop OS as understood by many people like those who forced MS to release Win 10 after Win 8 failure. It is a shame people behind Gnome - seemingly industry's WM of choice - do not understand that choice is what made Linux and limiting Gnome to support only their vision of UX prevents a wider Linux adoption.

Yes, above seems also off-topic but reasoning behind features of FOSS or even their mere existence should definitely be a part of development cycle.

**Comment by KaaCee** [comment]
For anyone who cannot wait for this feature (like myself ;) I added a pull request that adds the option "launch another" for the middle mouse button.
